two_population_analysis/Snakefile

```diff
@@ -1,83 +1,398 @@
+        output_dir +  "/{seeds}/dadi_analysis/{demo_models}/model_params_{fit_seeds}.txt"
+
+    threads: 4
+    run:
+        inputs = expand(output_dir + "/{seeds}/{chrms}_dadi_joint_sfs.fs", seeds=wildcards.seeds, chrms=chrm_list)
+        dadi_utils.fit_dadi_model(inputs,output[0],output[1],wildcards.demo_models, wildcards.fit_seeds,num_sampled_genomes_per_replicate)
+
+
+
+# ###############################################################################
+# fastsimcoal
+# ###############################################################################
+
+
+rule dadi_sfs_to_fsc:
+    # TODO: replace "IM" in the paths below with {demo_models}
+    input: expand(output_dir + "/{seeds}/{chrms}_dadi_joint_sfs_with_nonvariant.fs", chrms=chrm_list,seeds=seed_array)
+
+    output: output_dir + "/{seeds}/fsc_analysis/{demo_model}_jointDAFpop1_0.obs"
+
+    threads: 2
+    run:
+        inputs = expand(output_dir + "/{seeds}/{chrms}_dadi_joint_sfs_with_nonvariant.fs", chrms=chrm_list,seeds=wildcards.seeds)
+        dadi_out_path = output_dir + "/" + wildcards.seeds + "/all_chrm_dadi_joint_sfs_with_nonvariant.fs"
+        fsc_out_path  = output_dir + "/" + wildcards.seeds + "/fsc_analysis/"+wildcards.demo_model+"_jointDAFpop1_0.obs"
+        fastsimcoal.dadi_to_fsc_sfs(inputs, dadi_out_path, fsc_out_path, sample_size=num_sampled_genomes_per_replicate)
+
+
+rule fsc_setup:
+    input: expand(output_dir + "/{seeds}/fsc_analysis/{demo_model}_jointDAFpop1_0.obs", seeds=seed_array,demo_model=fsc_model)
+
+    output: output_dir + "/{seeds}/fsc_analysis/{demo_model}.est",
+            output_dir + "/{seeds}/fsc_analysis/{demo_model}.tpl"
+#            sed -i '' 's/SAMPLE_SIZE/{num_sampled_genomes_per_replicate}/g' {output_dir}/{wildcards.seeds}/fsc_analysis/IM.tpl &&\
+
+    shell: "cp fsc_files/{wildcards.demo_model}.est {output_dir}/{wildcards.seeds}/fsc_analysis/{wildcards.demo_model}.est && \
+            cp fsc_files/{wildcards.demo_model}.tpl {output_dir}/{wildcards.seeds}/fsc_analysis/{wildcards.demo_model}.tpl && \
+            sed -i.bak 's/SAMPLE_SIZE/{num_sampled_genomes_per_replicate}/' {output_dir}/{wildcards.seeds}/fsc_analysis/{wildcards.demo_model}.tpl && rm {output_dir}/{wildcards.seeds}/fsc_analysis/{wildcards.demo_model}.tpl.bak"
+
+
+rule run_fsc:
+    input: expand(output_dir + "/{seeds}/fsc_analysis/{demo_model}.tpl", seeds=seed_array,demo_model=fsc_model)
+
+    output: output_dir + "/{seeds}/fsc_analysis/run10/{demo_model}/{demo_model}.bestlhoods"
+
+
+    # I want to run this with a bash loop from 1..fit_runs to automate the number of fsc runs but for the life of me I cant get it to work in snakemake (it works fine in a shell script)
+    # it keeps spitting out a python 'out of index' error and I've spent way too much time trying to troubleshoot it with no success
+    # if someome can figure this out they would be my hero. Until that day, will continue to use this wall of bash code
```

Comment:
yeah this stuff is super inefficient, because you aren't letting snakemake run these things in parallel. once we merge i can fix this

Reply:
ugh sorry I know this is pretty gnarly, thanks for the help here